packages/data/src/index.ts

```diff
@@ -0,0 +1,31 @@
+import type { DataDriver } from './types/driver.js';
+import type { AbstractQuery } from './types/abstract-query.js';
+
+export class DataEngine {
```

Comment:
maybe we should make sure that this class can be initialized only once wherever it is used, so making it a singleton probably [like this](https://refactoring.guru/design-patterns/singleton/typescript/example#example-0--index-ts)